docs/proposals/20210323-etcdadm-bootstrap-provider.md

```diff
@@ -0,0 +1,556 @@
+    namespace: "default"
+  managedExternalEtcdRef:
+    kind: EtcdCluster
+    apiVersion: etcdcluster.cluster.x-k8s.io/v1alpha4
+    name: "my-cluster-etcd-cluster"
+    namespace: "default"
+---
+kind: EtcdCluster
+apiVersion: etcdcluster.cluster.x-k8s.io/v1alpha4
+metadata:
+  name: "my-cluster-etcd-cluster"
+  namespace: default
+spec:
+  etcdadmConfigSpec:
+    version: 3.4.13 #optional
+    # etcdadmArgs:
+    # users:
+  replicas: 3
+  infrastructureTemplate:
+    kind: DockerMachineTemplate
+    apiVersion: infrastructure.cluster.x-k8s.io/v1alpha4
+    name: "etcd-plane"
+    namespace: "default"
+---
+apiVersion: infrastructure.cluster.x-k8s.io/v1alpha4
+kind: DockerCluster
+metadata:
+  name: "my-cluster"
+  namespace: "default"
+---
+kind: KubeadmControlPlane
+apiVersion: controlplane.cluster.x-k8s.io/v1alpha4
+metadata:
+  name: "abcd-control-plane"
+  namespace: "default"
+spec:
+  replicas: 3
+  infrastructureTemplate:
+    kind: DockerMachineTemplate
+    apiVersion: infrastructure.cluster.x-k8s.io/v1alpha4
+    name: "abcd-control-plane"
+    namespace: "default"
+  kubeadmConfigSpec:
+    clusterConfiguration:
+      controllerManager:
+        extraArgs: {enable-hostpath-provisioner: 'true'}
+      apiServer:
+        certSANs: [localhost, 127.0.0.1]
+      etcd:
+        external:
+          endpoints: []
+          caFile: "/etc/kubernetes/pki/etcd/ca.crt"
+          certFile: "/etc/kubernetes/pki/apiserver-etcd-client.crt"
+          keyFile: "/etc/kubernetes/pki/apiserver-etcd-client.key"
+    initConfiguration:
+      nodeRegistration:
+        criSocket: /var/run/containerd/containerd.sock
+        kubeletExtraArgs: {eviction-hard: 'nodefs.available<0%,nodefs.inodesFree<0%,imagefs.available<0%'}
+    joinConfiguration:
+      nodeRegistration:
+        criSocket: /var/run/containerd/containerd.sock
+        kubeletExtraArgs: {eviction-hard: 'nodefs.available<0%,nodefs.inodesFree<0%,imagefs.available<0%'}
+  version: "v1.20.2"
+```
+
+![etcdcluster-sequence](images/managed-etcd/etcdcluster-sequence.png)
+
+- Cluster.Spec will have a new field `ManagedExternalEtcdRef` of type `ObjectReference` (same as `ControlPlaneRef`). For using CAPI managed etcd cluster, user will set this field in Cluster manifest as shown above.
+- Cluster controller's `reconcileControlPlane` will check if `managedExternalEtcdRef` is set on cluster spec and if External etcd cluster is not `Ready`, it will pause the controlPlane provisioning by setting clusterv1.Paused annotation (`cluster.x-k8s.io/paused`) on it. This will allow etcdcluster provisioning to occur first. 
+- Etcd cluster controller will generate a CA cert-key pair to be used for all etcd members. This external etcd CA will be saved in a Secret with name `{cluster.Name}-managed-etcd`. We can use the existing util/secret pkg in CAPI for this and add a new [Purpose](https://github.com/mrajashree/cluster-api/blob/etcdadm_bootstrap/util/secret/consts.go#L50) for managed etcd.
```

Comment:
Again do we define somewhere what is `Etcd cluster controller`? Maybe better if we explicitly call out some glossaries?